Src/IronPython/Runtime/Operations/PythonOps.cs

```diff
@@ -3608,6 +3605,15 @@ public static partial class PythonOps {
             return (EncoderFallbackException)ctor.Invoke(new object[] { message, charUnknownHigh, charUnknownLow, index });
         }
 
+        public static Exception UnicodeEncodeError(string message, int runeUnknown, int index) {
```

Comment:
Maybe it is a good time to ask: what are the rules to decide when a method in this assembly is made public or internal? 

I made this new overload public, because the other members in this method group are public, so for consistency. But technically, none of them need to be public. This is also true for many other exception factories, and yet the vast majority of them are public.

Reply:
Well, my preference is to make everything internal unless it needs to be public. The reason being the smaller the public API surface is the easier it is to evolve and maintain the codebase without having to worry about breaking changes. With IronPython 3 being a "new" project it's probably a good time to try and minimize the public API surface.

Examples of things that need(?) to be public:
- Methods that are used in expression trees and may end up in compiled code. Typically these (at least the methods) appear in `IronPython.Compiler.Ast.AstMethods` but there are probably some that are used directly?
- Python types (obviously)
- Things that would need to be exposed so someone could properly implement their own Python types and modules

Another thing that has been nagging at me is `PythonOps` being a sort of grab bag of things as opposed to a class that:
> Contains functions that are called directly from generated code to perform low-level runtime functionality.

I guess creating .NET exceptions could be considered "low-level runtime functionality" so maybe this is fine. Although I could also imagine having a separate class that would be the goto place for creating exceptions. I don't really have a good solution to this.

Reply:
OK, by these rules the new overload should be internal, since it is not really needed for anybody to " properly implement their own Python types and modules". Unlike the other two overloads, which provide access to a non-public constructor. Another reason to have it internal is that if the signature changes to use `Rune`, I'd rather not make our home-made Rune knock-off public.

I leave other exception factories as they are as it is outside the scope of this PR.